vic/drivers/classic/src/get_force_type.c

```diff
@@ -147,3 +147,50 @@ get_force_type(char *cmdstr,
 
     (*field)++;
 }
+
+
+/******************************************************************************
+ * @brief    This routine determines the counts the number of forcing variables
+             in each forcing file specified in the global parameter file.
+ *****************************************************************************/
+size_t
+count_force_vars(FILE *gp)
+{
+    size_t        nvars;
+    unsigned long start_position;
+    char          cmdstr[MAXSTRING];
+    char          optstr[MAXSTRING];
+    // Figure out where we are in the input file
+    fflush(gp);
```

Comment:
This command is no op if the stream (`gp`) does not have any buffered data.  If it does have buffered data, we want to flush it now so we get an accurate position of where we left off in `get_global_param()`.

The docs on `fflush()`:

       For input streams associated with seekable files (e.g., disk files,
       but not pipes or terminals), fflush() discards any buffered data that
       has been fetched from the underlying file, but has not been consumed
       by the application.